keps/sig-network/2030-topology-aware-proxying/README.md

```diff
@@ -245,6 +253,138 @@ not reliant on any other feature being enabled in any other release. [KEP
 #2004](https://github.com/kubernetes/enhancements/issues/2004) will be dependent
 on this KEP though.
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: TopologyAwareProxying
+    - Components depending on the feature gate: kube-proxy (also used by KEP
+      2004 for kube-controller-manager)
+
+* **Does enabling the feature change any default behavior?**
+  Not really. Kube-Proxy will still consume endpoints without these topology
+  labels (the default). When these labels are set (a separate opt-in discussed
+  in KEP 2004), kube-proxy will only consume a subset of endpoints closest to
+  it. This will also not be noticeable in most cases.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, disabling the feature gate or simply disabling this functionality per
+  Service (KEP 2004) will be supported.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  A potentially smaller set of endpoints may be routed to if this feature was
+  also enabled as described in KEP 2004.
+
+* **Are there any tests for feature enablement/disablement?**
+  There are not, but there will be tests covering the state of this feature
+  when it is and is not enabled. There is nothing significant involved in the
+  transition process, just turning the feature gate on or off.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
```

Comment:
Should we arrange kube-proxy to detect that there are no endpoints for "my" zone and fall back on another for-zone?  It would be a real change in how the watches were set up and would erase some of the optimization, but it offers some deeper defense against problems.

Reply:
I'm personally not treating this enhancement as scalability optimization, but rather as a feature for cost reduction, potential better within-zone latency, etc.

So I would be fine with still watching everything for the advantage of better defense.

Reply:
Yeah that makes sense. I think we'd still end up with a pretty significant win if kube-proxy is managing less iptables rules even if it is still tracking all EndpointSlices. This approach does seem like a safer way to start, I'll update the KEP to reflect that.

Reply:
+1 - if we would end up with e.g. 3x smaller iptables, that would be a huge win by itself.

Reply:
Yep, I think that's a much better solution, updated.